wasm/src/debug_service.rs

```diff
@@ -267,137 +262,130 @@ impl From<StepResultId> for usize {
     }
 }
 
-#[wasm_bindgen]
-#[derive(Serialize, Deserialize)]
-struct StructStepResult {
-    pub id: usize,
-    pub value: usize,
-}
-
-#[wasm_bindgen(typescript_custom_section)]
-const IStructStepResult: &'static str = r#"
-export interface IStructStepResult {
-    id: number;
-    value: number;
-}
-"#;
-
-#[wasm_bindgen(typescript_custom_section)]
-const IBreakpointSpanList: &'static str = r#"
-export interface IBreakpointSpan {
-    id: number;
-    lo: number;
-    hi: number;
-}
-
-export interface IBreakpointSpanList {
-    spans: Array<IBreakpointSpan>
-}
-"#;
-
-#[derive(Serialize, Deserialize)]
-struct BreakpointSpanList {
-    pub spans: Vec<BreakpointSpan>,
-}
-
-#[derive(Serialize, Deserialize, Copy, Clone, Debug, Eq, Hash, PartialEq)]
-struct BreakpointSpan {
-    pub id: u32,
-    pub lo: u32,
-    pub hi: u32,
-}
-
-#[wasm_bindgen(typescript_custom_section)]
-const IStackFrameList: &'static str = r#"
-export interface IStackFrame {
-    name: string;
-    path: string;
-    lo: number;
-    hi: number;
-}
-
-export interface IStackFrameList {
-    frames: Array<IStackFrame>
-}
-"#;
-
-#[derive(Serialize, Deserialize)]
-struct StackFrameList {
-    pub frames: Vec<StackFrame>,
+serializable_type! {
+    struct StructStepResult {
+        pub id: usize,
+        pub value: usize,
+    },
+    r#"export interface IStructStepResult {
+        id: number;
+        value: number;
+    }"#,
+    StructStepResult,
+    IStructStepResult,
+    "IStructStepResult"
```

Comment:
It's a little unfortunate you still need to repeat the name 'StructStepResult' 5 times when using the macro. No way to avoid this?

Reply:
I thought I couldn't, but now that I'm looking with fresh eyes I think I can eliminate at least one! Revising...